pkg/pipelinerunmetrics/metrics.go

```diff
@@ -409,6 +455,13 @@ func (r *Recorder) RunningPipelineRuns(lister listers.PipelineRunLister) error {
 					prsWaitResolvingPipelineRef++
 				}
 			}
+		} else {
+			// If all PipelineRuns for a Pipeline are completed then record it with 0
+			// set the key to 0 so that we do not record further completed PipelineRuns  for same Pipeline
+			if _, exists := countMap[pipelineRunKey]; !exists {
+				countMap[pipelineRunKey] = 0
```

Comment:
The `countMap` is only complete at the end of the loop, so why is the metric reported here?

Reply:
same reason as above.
this code is reset the metric already reported.


lets assume we have 4 PRs running for a key at given point in time so metric is reported with 4.
in next call to function all PR are completed so they wont be reported in in block which mean their metric report will still be showing 4 on graph which is wrong data.

but resetting the value to 0 here solves the issues
if there is any running PR for the key then that will be handled by if block. if not then anyways we are setting to 0.